taprpc/marshal.go

```diff
@@ -692,3 +692,37 @@ func MarshalAsset(ctx context.Context, a *asset.Asset,
 
 	return rpcAsset, nil
 }
+
+// ParseScriptKeyTypeQuery parses the script key type query from the RPC
+// variant.
+func ParseScriptKeyTypeQuery(
+	q *ScriptKeyTypeQuery) (fn.Option[asset.ScriptKeyType], bool, error) {
+
+	if q == nil || q.Type == nil {
+		return fn.Some(asset.ScriptKeyBip86), false, nil
+	}
+
+	switch t := q.Type.(type) {
+	case *ScriptKeyTypeQuery_ExplicitType:
+		explicitType, err := UnmarshalScriptKeyType(t.ExplicitType)
+		if err != nil {
+			return fn.None[asset.ScriptKeyType](), false, err
+		}
+
+		includeSpent := false
+		switch explicitType {
+		case asset.ScriptKeyTombstone, asset.ScriptKeyBurn:
+			includeSpent = true
```

Comment:
Why do we always want to include spent for burns and tombstones?

Reply:
Ah I see here from the proto comments, can add there here to imo: 
```
If the type
    // is set to SCRIPT_KEY_BURN or SCRIPT_KEY_TOMBSTONE the include_spent flag
    // will automatically be set to true, because assets of that type are always
    // marked as spent.
```

Reply:
Good point. Added that comment.